packages/core/content-manager/admin/src/services/api.ts

```diff
@@ -9,6 +9,8 @@ const contentManagerApi = adminApi.enhanceEndpoints({
     'InitialData',
     'HistoryVersion',
     'Relations',
+    'Release',
+    'ReleaseAction',
```

Comment:
The problem is that changes to documents could cause stale release data. For example, if you have an invalid entry that causes a release to have a "Blocked" status, and then you go to the Edit View and update the entry to fit the validation rules, this changes the release status to "Ready". This is the scenario I want to address here.

Reply:
you should be doing it this way in that case i think? https://redux-toolkit.js.org/rtk-query/api/created-api/code-splitting#enhanceendpoints

Reply:
I think I didn't explain the first question 😅 I'm doing that I think, for example I invalidate the releases tags [here](https://github.com/strapi/strapi/blob/62ce06180fe9a772eaeb3d43d238b26644f39f7c/packages/core/content-manager/admin/src/services/documents.ts#L338) on the update document.

Adding the types here is to avoid a Typescript error that not allowed me to invalidate the releases tags because they are not known. [Here](https://github.com/reduxjs/redux-toolkit/issues/1510#issuecomment-1546751373) is a discussion about this on the RTK's repo

Reply:
you're invalidating it in the wrong place, because thats the endpoint definition in the domain of the content-manager, instead your `enhanceEndpoints` object on the CR service should look like this:

```ts
  .enhanceEndpoints({
    addTagTypes: ['Release', 'ReleaseAction', 'EntriesInRelease'],
    endpoints: {
      updateDocument: {
        invalidatesTags: [
          { type: 'Release', id: 'LIST' },
          { type: 'ReleaseAction', id: 'LIST' },
        ],
      },
      deleteDocument: {
        invalidatesTags: [
          { type: 'Release', id: 'LIST' },
          { type: 'ReleaseAction', id: 'LIST' },
        ],
      },
      deleteManyDocuments: {
        invalidatesTags: [
          { type: 'Release', id: 'LIST' },
          { type: 'ReleaseAction', id: 'LIST' },
        ],
      },
      discardDocument: {
        invalidatesTags: [
          { type: 'Release', id: 'LIST' },
          { type: 'ReleaseAction', id: 'LIST' },
        ],
      },
    },
  })
```

where you add the invalidate tags from this side, keeping the release logic locked away. This is the way i'd expect plugin authors to have their data invalidated when a document is published, for example.

Reply:
thanks! This works way better